src/Config.h

```diff
@@ -42,14 +42,17 @@ class Config {
 
 		bool ServerEnabled;
 		uint64_t ServerPort;
+		uint64_t PersistentDataMaxSize;
 
 		bool DebugEnabled;
 		bool AlertsEnabled;
+		bool PersistentDataEnabled;
 
 		bool AuthRequired;
 		QString Secret;
 		QString Salt;
 		QString SessionChallenge;
+		QString PersistentDataStore;
```

Comment:
In-memory, I'd rather have the data store be a QJsonDocument, and have it serialized on config save and deserialized on config load. Deserializing and serializing the whole store on every change (request) has a huge performance cost.

Reply:
Well this is what you requested before :)

> Regarding how data is stored, I'm not exactly a fan of using several files for each realm. I'd be more comfortable with storing everything into the same spot: a QJsonObject instance that gets serialized on load and deserialized on save into a global config key. 

I'm unsure how to do what you ask for now. If I don't serialize on change then the change would not persist when for example OBS crashes or am I missing something?

Reply:
I think the confusion lies here: "deserialized on load and serialized on save" (opposite of what I wrote btw). To be more precised, it's deserialized on config load and serialized on config save, so it exists as a QJsonObject in the Config object.

Reply:
It would be better to use QJsonObject in the Config Object, it is created from the QJSonDocument on the fly and would break reentrancy (see other comment). I do not see any downside of directly putting the QJsonObject in the config object?